In-memory suggestions

hubertp opened the pull request:
### Pull Request Description

This change replaces an sqllite-backed suggestions' repo with a simple, in-memory, one.
As `completion` functionality has been implemented completely in GUI, there is no need to support it in backend, which simplifies a lot of functionality.

Closes #9650 and #9471.

### Important Notes

Loading suggestions and sending them to GUI on startup is almost instantaneous. Previously it would take ~10s just for `Standard.Base`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

hubertp committed:
Towards in-memory suggestions

hubertp committed:
Tests compile

Ideally would like to use `SuggestionsRepo[Future]` instead of having a
concrete implementation but mixing Scala and Java will mean that Java
expects `SuggestionsRepo<Future<?>>` i.e. a conrete type rather than a
type constructor which breaks everything. This also explains why
previously we had `SqlSuggestionsRepo` all over the place. So much for
nice abstractions.

hubertp committed:
remove completions endpoint

hubertp committed:
Remove unnecessary clutter, API complete

hubertp committed:
fix remaining test

hubertp committed:
don't bump version on clean action

hubertp committed:
nit

hubertp committed:
update  benchmark

JaroslavTulach commented on lib/scala/searcher/src/bench/java/org/enso/searcher/sql/SuggestionsRepoBenchmark.java:
Is anyone running, checking results of this benchmark?

hubertp replied:
:shrug:

Akirathan replied:
Nope. We don't have any historical data about this benchmark anywhere. The reason being that it is in `lib/scala/searcher` project. I guess that it does not make sense to start collecting the data now anymore.

hubertp replied:
> I guess that it does not make sense to start collecting the data now anymore.

Nope

JaroslavTulach commented on engine/polyglot-api/src/main/scala/org/enso/polyglot/Suggestion.scala:
```diff
@@ -332,6 +377,26 @@ object Suggestion {
+    override def withReturnType(returnType: String): Suggestion =
+      copy(returnType = returnType)
+
+    override def update(
```
Why is this method completely the same as the same method on line 476?

hubertp replied:
There is a small amount of DRY in this case but I would prefer to keep it that way rather than overriding a copy method generated for case-classes.

JaroslavTulach commented on engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/BaseServerTest.scala:
```diff
@@ -162,8 +162,7 @@ abstract class BaseServerTest
 
   val zioExec         = ZioExec(zioRuntime)
-  val sqlDatabase     = SqlDatabase(config.directories.suggestionsDatabaseFile)
-  val suggestionsRepo = new SqlSuggestionsRepo(sqlDatabase)(system.dispatcher)
```
`SqlSuggestionsRepo` isn't removed by this PR. Is it intended to be used somehow?

hubertp replied:
The PR is still in a draft state...

JaroslavTulach reviewed (approved):
If the IDE gets suggestions faster, then everything is great!

hubertp committed:
Bump version only on change

Bringing back the old expectations in tests.

hubertp committed:
nit

hubertp committed:
Remove dependency on slick/jdbc

Removing the dependency revealed some unintended usage in tests.
Switching fully to the inmemory version also reveaeled some
implementation issues.

hubertp committed:
fix bench

hubertp committed:
legal review update

hubertp commented on build.sbt:
```diff
@@ -2169,14 +2167,15 @@ lazy val `engine-runner` = project
+      "org.typelevel"          %% "cats-core"               % catsVersion,
+      "junit"                   % "junit"                   % junitVersion              % Test,
+      "com.github.sbt"          % "junit-interface"         % junitIfVersion            % Test,
+      "org.scala-lang.modules" %% "scala-collection-compat" % scalaCollectionCompatVersion
```
note: `scala-collection-compat` was a dependency coming from slick stuff anyway. We seem to rely on it in the runner.

hubertp committed:
nit
